src/Tribe/Main.php

```diff
@@ -1394,6 +1396,81 @@ public function fix_all_day_events( $old_value, $new_value ) {
 			$wpdb->query( $fix_end_dates );
 		}
 
+		/**
+		 * Selects events to be moved to trash or permanently deleted.
+		 *
+		 * @see 'update_option_'.Tribe__Main::OPTIONNAME
+		 */
+		public function select_events_to_delete( $field_name, $old_value, $new_value ) {
+			$default_value = null;
+
+			// If the field value is changed to 'Disabled', we can exit the execution here
+			if ( $new_value[ $field_name ] == $default_value ) {
+				return;
+			}
+			// If the value is not changed, we can exit the execution here
+			if ( $old_value[ $field_name ] == $new_value[ $field_name ] ) {
+				return;
+			}
+
+			global $wpdb;
+
+			$event_month_cutoff = $new_value[ $field_name ];
+
+			$sql = "SELECT post_id
+					FROM {$wpdb->posts} AS t1
+					INNER JOIN {$wpdb->postmeta} AS t2 ON t1.ID = t2.post_id
+					WHERE t1.post_type = %s
+						AND t2.meta_key = '_EventEndDate'
+						AND t2.meta_value <= DATE_SUB( CURDATE(), INTERVAL %s MONTH )";
+
+			$args = [
+				'post_type'  => self::POSTTYPE,
+				'meta_value' => $event_month_cutoff,
+			];
+
+			$sql  = apply_filters( 'tribe_events_delete_old_events_sql', $sql );
+			$args = apply_filters( 'tribe_events_delete_old_events_sql_args', $args );
+
+			// Returns an array of Post IDs (events) that ended before a specific date
+			$post_ids = $wpdb->get_col( $wpdb->prepare( $sql, $args ) );
+
+			return $post_ids;
+		}
+
+		/**
+		 * Moves to trash events that ended before a date specified by user
+		 */
+		public function move_past_events_to_trash( $old_value, $new_value ) {
```

Comment:
Add docblock params:
- `@since TBD` for when we added
- `@param` for all the params
- `@return` for which kind of return